src/lib/RpcApi.ts

```diff
@@ -98,17 +103,29 @@ export default class RpcApi {
             undefined, // preserveRequests: keep default behavior, which is true for redirects but false for postMessage
             handleHistoryBack,
         );
+        if (!handleHistoryBack) {
+            // The Keyguard client rejects on history back only if handleHistoryBack is activated. If the Keyguard does
+            // not reject it also does not provide us the localState to recover. For this case, we encode it manually in
+            // the history, to retrieve it from there.
+            this._exportStateToHistory();
+        }
         return client;
     }
 
-    public routerPush(routeName: string) {
-        const query = this._parseUrlParams(window.location.search);
-        this._router.push({name: routeName, query});
+    public routerPush(routeName: string, query?: { [key: string]: string }) {
+        query = query || this._parseUrlParams(window.location.search);
+        this._router.push({name: routeName, query}, () => {
+            // export state to the newly pushed history entry to be available on reload
+            this._exportStateToHistory();
```

Comment:
So spamming the browser's history storage with JSON objects on every navigation is required to be able to reload at all times?

Reply:
Also, the callback is the `onComplete` callback here and gets called after the navigation is complete. I believe it makes more sense to export the state _before and after_ the navigation: the latest state still in the previous history entry - and also first thing in the new screen, later being replaced by the latest state when navigating away. That way the latest state, and all the changes made to the state in that screen, are preserved in its history entry, not in the next. 

Right now this saves a state to the history, which then changes in the screen (because that's what the UI is for, to change the state), but that state then only gets saved in the _next_ navigation entry.

Am I thinking right?

Reply:
See https://github.com/nimiq/hub/pull/364#discussion_r366095837 for an answer related to your first question.
Firefox imposes a limit of 640k characters per history state and we're way below that limit, so I guess we're not super wasteful here.

The question about when to export the state to the history is a design decision.
Storing stuff in the static store is mostly done as input for the _next_ page. Therefore the data is only persisted in the state for the next page.
We can discuss whether there are use cases for updating the store also for the current page.

Reply:
I discussed this here with Sebastian and we decided that for the moment the current implementation makes more sense:
- The static store is mostly used to set input for the next page.
- Restoring the static store while other parts of the page are not restored (inputs, etc.) might be error prone or lead to confusion by the user.
- As the static store is not reactive, restoring a state might be undesirable as later changes will not lead to reactive updates.